kairos-contracts/demo-contract/contract/src/main.rs

```diff
@@ -81,19 +88,69 @@ pub extern "C" fn deposit() {
     casper_event_standard::emit(new_deposit_record);
 }
 
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct Proof {
+    pub receipt: risc0_zkvm::Receipt,
+    pub program_id: [u32; 8],
+}
+
+#[no_mangle]
+pub extern "C" fn submit_batch() {
+    let proof_serialized: Vec<u8> = runtime::get_named_arg(RUNTIME_ARG_RECEIPT);
+    let proof: Proof = from_slice(&proof_serialized).unwrap();
```

Comment:
NIT: makes it clearer where `from_slice` comes from.
```suggestion
    let proof: Proof = serde_json_wasm::from_slice(&proof_serialized).unwrap();
```